### PR TITLE
[WIP][9.0] Add sale_automatic_workflow_payment_mode

### PR DESCRIPTION
**Pull request by guewen**
Glue module between sale_automatic_workflow and account_payment_sale

Depends on #291 and https://github.com/OCA/bank-payment/pull/270

### CODE REVIEW

**Comment by sebastienbeau** [comment]
:+1: Great we finaly converge ! code review ok

**Comment by atchuthan** [comment]
code reviewed :+1:

**Comment by florian-dacosta** [comment]
code review :+1:

**Comment by atchuthan** [comment]
@guewen As https://github.com/OCA/bank-payment/pull/270 is already merged, I tried rebuilding the runbot, but it seems to be failing due to missing **bank-payment** in oca_dependencies.txt.

**Comment by atchuthan** [comment]
@guewen sale_automatic_workflow is not in the working branch of this PR(https://github.com/guewen/sale-workflow/tree/add-sale_automatic_workflow_payment_mode), thus failing to find the module on runbot test.

**Log from ci/runbot test:**
2016-07-25 13:23:28 INFO    runbot  odoo.runbot:0 init Init build environment
2016-07-25 13:23:57 INFO    runbot  odoo.runbot:0 test_base Start test base module
2016-07-25 13:24:15 ERROR   server  openerp.modules.graph:114 add_modules module sale_automatic_workflow_payment_mode: Unmet dependencies: sale_automatic_workflow, account_payment_sale
2016-07-25 13:41:32 WARNING server  openerp.modules.graph:118 add_modules Some modules were not loaded.
2016-07-25 13:40:45 INFO    runbot  odoo.runbot:0 run Start running build 3151072-297-95a002

**Comment by atchuthan** [comment]
@guewen: any comment about the last 2 discussions/comments.

**Comment by guewen** [comment]
@atchuthan I rebased

**Comment by guewen** [comment]
Added a dependency on the repository oca/bank-payment

**Comment by parthivgls** [comment]
:+1:  can we move ahead with this PR ? as it's blocking https://github.com/OCA/connector-ecommerce/pull/32

File: sale_automatic_workflow_payment_mode/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# © 2016 Camptocamp SA, Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+{'name': 'Sale Automatic Workflow - Payment Mode',
+ 'version': '9.0.2.0.0',
```
Comment (pedrobaeza) [inline sale_automatic_workflow_payment_mode/__openerp__.py:6]: Why 9.0.2.0.0?

File: sale_automatic_workflow_payment_mode/data/automatic_workflow_data.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+    sale_automatic_workflow_payment_mode for Odoo
+    Copyright (C) 2016 Sodexis
+    The licence is in the file __openerp__.py
+-->
+<odoo>
+    <data noupdate="1">
+        <record id="automatic_workflow_payment_filter" model="ir.filters">
```
Comment (pedrobaeza) [inline sale_automatic_workflow_payment_mode/data/automatic_workflow_data.xml:9]: Why the filter is under noupdate=1 section?

File: sale_automatic_workflow_payment_mode/models/automatic_workflow_job.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    @api.model
+    def _register_payments(self, payment_filter):
+        invoice_obj = self.env['account.invoice']
+        invoices = invoice_obj.search(payment_filter)
+        _logger.debug('Invoices to Register Payment: %s', invoices.ids)
+        for invoice in invoices:
+            partner_type = invoice.type in ('out_invoice', 'out_refund') and \
+                'customer' or 'supplier'
+            payment_mode = invoice.payment_mode_id
+            with commit(self.env.cr):
```
Comment (pedrobaeza) [inline sale_automatic_workflow_payment_mode/models/automatic_workflow_job.py:36]: Why is this block inside an auto-commit?

Comment (guewen) [reply sale_automatic_workflow_payment_mode/models/automatic_workflow_job.py:36]: This is how `sale_automatic_workflow` works, every individual action is directly committed so even if something fails, the rest stays.

File: sale_automatic_workflow_payment_mode/data/automatic_workflow_data.xml (hunk 2 above)
Comment (guewen) [reply sale_automatic_workflow_payment_mode/data/automatic_workflow_data.xml:9]: Because the user is allowed to change the filter.

**Comment by pedrobaeza** [comment]
OK, we can move on with this although the version seems a bit strange, but it's not a blocker.

**Pull request merged**